### PR TITLE
[galactic] Fix quiet filtering for node_topic graphs

### PR DESCRIPTION
**Pull request by DLu**
## New version of PR #69 based on galactic-devel

Here's a standard graph with Debug/Quiet=False

Screenshot from 2021-09-28 10-44-57

However, when we turn on Debug/Quiet, we are not correctly filtering the edges.
Screenshot from 2021-09-28 10-44-53
 
This is an effect of the nodes being properly filtered, but the edges not being filtered, so new nodes with the dot node names (`t__rosout`) are created. 

The new logic always filters the edges (instead of just when `graph_mode == NODE_NODE_GRAPH`. We also always filter the `nt_nodes`, which is just empty if `graph_mode == NODE_NODE_GRAPH`. 

@hidmic I think you wrote this bit of logic originally in #35

### CODE REVIEW

**Review by hidmic — APPROVED** [review]
Tested locally. LGTM pending green CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15357/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/10067/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/13043/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15561/)

**Comment by hidmic** [comment]
Alright, going in! Thanks for the fix @DLu.

**Comment by hidmic** [comment]
@DLu would you mind to sign off your commit before I merge.

**Comment by DLu** [comment]
Done

**Comment by DLu** [comment]
@hidmic How do you want to backport? I can open up new PRs if you want

**Comment by hidmic** [comment]
> How do you want to backport? I can open up new PRs if you want

A backport PR for Foxy sounds good. We can run Foxy CI there.

**Pull request merged**
